x/deployment/client/cli/tx.go

```diff
@@ -32,10 +40,152 @@ func GetTxCmd(key string, cdc *codec.Codec) *cobra.Command {
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Setup CLI context and transaction builder
+			ctx := context.NewCLIContext().WithCodec(cdc)
+			bldr := auth.NewTxBuilderFromCLI(os.Stdin).WithTxEncoder(utils.GetTxEncoder(cdc))
+			if err := ctx.Client.Start(); err != nil {
+				return err
+			}
+
+			// Read in manifest file
+			sdl, err := sdl.ReadFile(args[0])
+			if err != nil {
+				return err
+			}
+
+			// Fetch deployment groups from the SDL file, each group maps to an individual order
+			groups, err := sdl.DeploymentGroups()
+			if err != nil {
+				return err
+			}
+
+			// Fetch the manifest from the SDL file
+			mani, err := sdl.Manifest()
+			if err != nil {
+				return err
+			}
+
+			// Get the deployment ID (defaults to height)
+			id, err := DeploymentIDFromFlags(cmd.Flags(), ctx.GetFromAddress().String())
+			if err != nil {
+				return err
+			}
+
+			// Default DSeq to the current block height
+			if id.DSeq == 0 {
+				if id.DSeq, err = currentBlockHeight(ctx); err != nil {
+					return err
+				}
+			}
+
+			// create a new pubsub bus to handle chain events
+			bus := pubsub.NewBus()
+			defer bus.Close()
+
+			// Create an error group to handle chain listener
+			group, cctx := errgroup.WithContext(ccontext.Background())
+
+			// Listen to the events on chain and publish them to the pubsub bus
+			group.Go(func() error {
+				return events.Publish(cctx, ctx.Client, "deployment-create", bus)
+			})
+
+			// expose event channel
+			sub, err := bus.Subscribe()
+			if err != nil {
+				return err
+			}
+
+			// start goroutine to listen for events
+			leases := len(groups)
+			group.Go(func() error {
+				for {
+					select {
+					case <-sub.Done():
+						return nil
+					case ev := <-sub.Events():
+						switch msg := ev.(type) {
+						case types.EventDeploymentCreated:
+							fmt.Printf("Deployment %d created...\n", msg.ID.DSeq)
+						case mtypes.EventOrderCreated:
+							fmt.Printf("Order %d for deployement %d created...\n", msg.ID.OSeq, msg.ID.DSeq)
+						case mtypes.EventBidCreated:
+							fmt.Printf("Bid of %s for order %d:%d created...\n", msg.Price, msg.ID.DSeq, msg.ID.OSeq)
+						case mtypes.EventLeaseCreated:
+							fmt.Printf("Lease for order %d:%d created...\n", msg.ID.DSeq, msg.ID.OSeq)
+							pclient := pmodule.AppModuleBasic{}.GetQueryClient(ctx)
+							provider, err := pclient.Provider(msg.ID.Provider)
+							if err != nil {
+								return err
+							}
+
+							gclient := gateway.NewClient()
+
+							fmt.Printf("Sending manifest to provider %s...\n", msg.ID.Provider)
+							if err = gclient.SubmitManifest(
+								ccontext.Background(),
+								provider.HostURI,
+								&manifest.SubmitRequest{
+									Deployment: msg.ID.DeploymentID(),
+									Manifest:   mani,
+								},
+							); err != nil {
+								return err
+							}
+
+							if leases = leases - 1; leases == 0 {
+								fmt.Printf("Leases left %d...\n", leases)
```

Comment:
Planning on exiting the command here, but unable to make it exit cleanly with the current code.

Reply:
have this exiting cleanly!